src/NuGet.Clients/NuGet.PackageManagement.UI/PackageLicenseUtilities.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Packaging;
+using NuGet.Packaging.Licenses;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.PackageManagement.UI
+{
+    internal class PackageLicenseUtilities
+    {
+
+        internal static IList<IText> GenerateLicenseLinks(DetailedPackageMetadata metadata)
+        {
+            return GenerateLicenseLinks(metadata.LicenseMetadata, metadata.LicenseUrl);
+        }
+
+        internal static IList<IText> GenerateLicenseLinks(IPackageSearchMetadata metadata)
+        {
+            return GenerateLicenseLinks(metadata.LicenseMetadata, metadata.LicenseUrl);
+        }
+
+        private static IList<IText> GenerateLicenseLinks(LicenseMetadata licenseMetadata, Uri licenseUrl)
+        {
+            IList<IText> list = new List<IText>();
+            if (licenseMetadata != null)
+            {
+                list = GenerateLicenseLinks(licenseMetadata);
+            }
+            else
+            {
+                if (licenseUrl != null)
+                {
+                    list.Add(new LicenseText(Resources.Text_LicenseAcceptance, licenseUrl));
+                }
+            }
+            return list;
+        }
+
+        // Internal for testing purposes. // TODO NK - Add tests for file. 
+        internal static IList<IText> GenerateLicenseLinks(LicenseMetadata metadata)
+        {
+
+            var list = new List<IText>();
+            switch (metadata.Type)
+            {
+                case LicenseType.Expression:
+
+                    if (metadata.LicenseExpression != null)
+                    {
+                        var identifiers = new List<string>();
+                        GetLicenseIdentifiers(metadata.LicenseExpression, identifiers);
+
+                        var licenseToBeProcessed = metadata.License;
+
+                        foreach (var identifier in identifiers)
+                        {
+                            var licenseStart = licenseToBeProcessed.IndexOf(identifier);
+                            if (licenseStart != 0)
+                            {
+                                list.Add(new FreeText(licenseToBeProcessed.Substring(0, licenseStart)));
+                            }
+                            var license = licenseToBeProcessed.Substring(licenseStart, identifier.Length);
+                            list.Add(new LicenseText(license, new Uri($"https://spdx.org/licenses/{license}.html")));
```

Comment:
create a constant for the spdx url

Reply:
I was gonna change this before merging. 
It'll be our Url, I will create a constant for it regardless though :)